test/DurableTask.Sdk.Tests/Grpc/DurableTaskGrpcClientIntegrationTests.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.DurableTask.Tests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DurableTask.Grpc.Tests;
+
+public class DurableTaskGrpcClientIntegrationTests : IntegrationTestBase
+{
+    const string OrchestrationName = "TestOrchestration";
+
+    public DurableTaskGrpcClientIntegrationTests(ITestOutputHelper output, GrpcSidecarFixture sidecarFixture)
+        : base(output, sidecarFixture)
+    {
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task GetInstanceMetadata_EndToEnd(bool shouldThrow)
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, string instanceId, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().Be(instanceId);
+                metadata.SerializedInput.Should().BeNull();
+                metadata.SerializedOutput.Should().BeNull();
+                metadata.RuntimeStatus.Should().Be(status);
+                metadata.FailureDetails.Should().BeNull();
+            }
+        }
+
+        await using DurableTaskGrpcWorker server = await this.StartAsync();
+        DurableTaskClient client = this.CreateDurableTaskClient();
+
+        string instanceId = await client.ScheduleNewOrchestrationInstanceAsync(OrchestrationName, input: shouldThrow);
+
+        await client.WaitForInstanceStartAsync(instanceId, default);
+        OrchestrationMetadata? metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, OrchestrationRuntimeStatus.Running);
+
+        await client.RaiseEventAsync(instanceId, "event", default);
+        await client.WaitForInstanceCompletionAsync(instanceId, default);
+        metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, shouldThrow ? OrchestrationRuntimeStatus.Failed : OrchestrationRuntimeStatus.Completed);
+    }
+
+    [Fact]
+    public async Task GetInstances_EndToEnd()
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().StartWith("GetInstances_EndToEnd-");
+                metadata.RuntimeStatus.Should().Be(status);
+
+                // InMemoryOrchestrationService always returns these in a query.
+                // metadata.FailureDetails.Should().BeNull();
+                // metadata.SerializedInput.Should().BeNull();
+                // metadata.SerializedOutput.Should().BeNull();
```

Comment:
Do these need to be removed (or uncommented)?

Reply:
I have them there commented because I _would_ like to verify that behavior, but `InMemoryOrchestrationService` from the sidecar library always returns those, even though I have `FetchInputsAndOutputs = false`. The comment is a reminder incase anyone else tries to verify the same thing I did.

Reply:
Updated the comment and inverted the statement, so we will know when to correct the assertion.